frame/balances/src/lib.rs

```diff
@@ -602,1601 +468,646 @@ pub mod pallet {
-			self.0 = self.0.saturating_add(other.0);
-			mem::forget(other);
+			// First we try to modify the account's balance to the forced balance.
+			let old_free = Self::mutate_account_handling_dust(&who, |account| {
+				let old_free = account.free;
+				account.free = new_free;
+				old_free
+			})?;
 
-			self
-		}
-		fn subsume(&mut self, other: Self) {
-			self.0 = self.0.saturating_add(other.0);
-			mem::forget(other);
-		}
-		fn offset(self, other: Self::Opposite) -> SameOrOther<Self, Self::Opposite> {
-			let (a, b) = (self.0, other.0);
-			mem::forget((self, other));
-
-			if a > b {
-				SameOrOther::Same(Self(a - b))
-			} else if b > a {
-				SameOrOther::Other(PositiveImbalance::new(b - a))
-			} else {
-				SameOrOther::None
+			// This will adjust the total issuance, which was not done by the `mutate_account`
+			// above.
+			if new_free > old_free {
+				mem::drop(PositiveImbalance::<T, I>::new(new_free - old_free));
+			} else if new_free < old_free {
+				mem::drop(NegativeImbalance::<T, I>::new(old_free - new_free));
 			}
-		}
-		fn peek(&self) -> T::Balance {
-			self.0
-		}
-	}
 
-	impl<T: Config<I>, I: 'static> Drop for PositiveImbalance<T, I> {
-		/// Basic drop handler will just square up the total issuance.
-		fn drop(&mut self) {
-			<super::TotalIssuance<T, I>>::mutate(|v| *v = v.saturating_add(self.0));
-		}
-	}
-
-	impl<T: Config<I>, I: 'static> Drop for NegativeImbalance<T, I> {
-		/// Basic drop handler will just square up the total issuance.
-		fn drop(&mut self) {
-			<super::TotalIssuance<T, I>>::mutate(|v| *v = v.saturating_sub(self.0));
+			Self::deposit_event(Event::BalanceSet { who, free: new_free });
+			Ok(().into())
 		}
 	}
-}
-
-impl<T: Config<I>, I: 'static> Currency<T::AccountId> for Pallet<T, I>
-where
-	T::Balance: MaybeSerializeDeserialize + Debug,
-{
-	type Balance = T::Balance;
-	type PositiveImbalance = PositiveImbalance<T, I>;
-	type NegativeImbalance = NegativeImbalance<T, I>;
-
-	fn total_balance(who: &T::AccountId) -> Self::Balance {
-		Self::account(who).total()
-	}
 
-	// Check if `value` amount of free balance can be slashed from `who`.
-	fn can_slash(who: &T::AccountId, value: Self::Balance) -> bool {
-		if value.is_zero() {
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Ensure the account `who` is using the new logic.
+		///
+		/// Returns `true` if the account did get upgraded, `false` if it didn't need upgrading.
+		pub fn ensure_upgraded(who: &T::AccountId) -> bool {
+			let mut a = T::AccountStore::get(who);
+			if a.flags.is_new_logic() {
+				return false
+			}
+			a.flags.set_new_logic();
+			if !a.reserved.is_zero() || !a.frozen.is_zero() {
+				if !system::Pallet::<T>::can_inc_consumer(who) {
+					// Gah!! We have a non-zero reserve balance but no provider refs :(
+					// This shouldn't practically happen, but we need a failsafe anyway: let's give
+					// them enough for an ED.
+					a.free = a.free.min(T::ExistentialDeposit::get());
```

Comment:
`min` or `max`?
also does this just mint ED without update total issuance?

Reply:
Good catch on the `min`: https://github.com/paritytech/substrate/pull/13741

It doesn't update total issuance, but as per the comment: this is an emergency measure shich should never be called in practice.